src/components/dashboard/clickable_cell_store.ts

```diff
@@ -0,0 +1,86 @@
+import { markRaw } from "@odoo/owl";
+import { positionToZone, toXC } from "../../helpers";
+import { clickableCellRegistry } from "../../registries/cell_clickable_registry";
+import { SpreadsheetStore } from "../../stores/spreadsheet_store";
+import {
+  CellPosition,
+  Command,
+  Rect,
+  SpreadsheetChildEnv,
+  UID,
+  Zone,
+  invalidateEvaluationCommands,
+} from "../../types";
+
+type Garbage = ((position: CellPosition, env: SpreadsheetChildEnv) => void) | false;
+
+export interface ClickableCell {
+  coordinates: Rect;
+  position: CellPosition;
+  action: (position: CellPosition, env: SpreadsheetChildEnv) => void;
+}
+
+export class ClickableCellsStore extends SpreadsheetStore {
+  private _clickableCells: Record<UID, Record<string, Garbage>> = markRaw({});
+
+  handle(cmd: Command) {
+    if (
+      invalidateEvaluationCommands.has(cmd.type) ||
+      cmd.type === "EVALUATE_CELLS" ||
+      (cmd.type === "UPDATE_CELL" && cmd.content)
+    ) {
+      this._clickableCells = markRaw({});
+    }
+  }
+
+  private getClickableAction(position: CellPosition): Garbage {
+    const { sheetId, col, row } = position;
+    const clickableCells = this._clickableCells;
+    const xc = toXC(col, row);
+    if (!clickableCells[sheetId]) {
+      clickableCells[sheetId] = {};
+    }
+    if (!(xc in clickableCells[sheetId]!)) {
+      clickableCells[sheetId][xc] = this.findClickableAction(position);
+    }
+    return clickableCells[sheetId][xc];
+  }
+
+  private findClickableAction(position: CellPosition) {
+    const getters = this.getters;
+    for (const item of clickableCellRegistry.getAll().sort((a, b) => a.sequence - b.sequence)) {
```

Comment:
Probably a non-factor because the registry isn't big, be we create a new array + sort it at each call of this method

Reply:
it could be better 👍